addons/binding/org.openhab.binding.smappee/src/main/java/org/openhab/binding/smappee/internal/discovery/SmappeeDiscoveryService.java

```diff
@@ -0,0 +1,202 @@
+    private final Logger logger = LoggerFactory.getLogger(SmappeeDiscoveryService.class);
+    private SmappeeService _smappeeService;
+    private ThingUID _bridgeUID;
+
+    ScheduledFuture<?> scheduledJob;
+
+    /**
+     * Whether we are currently scanning or not
+     */
+    private boolean _scanning;
+
+    /**
+     * Constructs the discovery class using the thing IDs that smappee can discover
+     * - actuators (plugs)
+     * - detected appliances
+     */
+    public SmappeeDiscoveryService(SmappeeService smappeeService, ThingUID bridgeUID) {
+        super(ImmutableSet.of(SmappeeBindingConstants.THING_TYPE_ACTUATOR,
+                SmappeeBindingConstants.THING_TYPE_APPLIANCE), SEARCH_TIME, false);
+
+        _smappeeService = smappeeService;
+        _bridgeUID = bridgeUID;
+    }
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypes() {
+        return SmappeeBindingConstants.SUPPORTED_THING_TYPES_UIDS;
+    }
+
+    /**
+     * Activate the discovery service.
+     */
+    public void activate() {
+        super.activate(null);
+    }
+
+    /**
+     * Deactivate the discovery service.
+     */
+    @Override
+    public void deactivate() {
+        super.deactivate();
+    }
+
+    /**
+     * Starts the scan. This discovery will:
+     * <ul>
+     * <li>Call 'get service location info'
+     * (https://smappee.atlassian.net/wiki/spaces/DEVAPI/pages/8552482/Get+Servicelocation+Info)</li>
+     * <li>For each actuator, add a thing in the inbox (if it doesn't exist already)</li>
+     * <li>For each appliance, add a thing in the inbox (if it doesn't exist already)</li>
+     * </ul>
+     * The process will continue until {@link #stopScan()} is called.
+     */
+    @Override
+    protected void startScan() {
+        if (_scanning) {
+            stopScan();
+        }
+
+        // this should be done by base class ???
+        // somehow this is not working, so starting a scheduler instead
+        startAutomaticRefresh();
+
+        _scanning = true;
+    }
+
+    public void startAutomaticRefresh() {
+        Runnable runnable = () -> {
+            scanForNewDevices();
+        };
+
+        scheduledJob = scheduler.scheduleWithFixedDelay(runnable, 0, 5, TimeUnit.MILLISECONDS);
+    }
+
+    private void scanForNewDevices() {
+        if (!_smappeeService.isInitialized()) {
+            logger.debug("skipping discovery because smappee service is not up yet (config error ?)");
+            return;
+        }
+
+        logger.debug("Starting Discovery");
+
+        SmappeeServiceLocationInfo serviceLocationInfo = _smappeeService.getServiceLocationInfo();
+
+        if (serviceLocationInfo == null) {
+            logger.warn("failed to scan for new smappee devices");
+            return;
+        }
+
+        for (SmappeeServiceLocationInfoActuator actuator : serviceLocationInfo.actuators) {
+            String id = actuator.id;
+            ThingTypeUID typeId = SmappeeBindingConstants.THING_TYPE_ACTUATOR;
```

Comment:
Inline it, you don't need this variable pointing to a constant.

Reply:
Ok